### ISSUE TITLE
Unused value analyzer false positive with ??=

### ISSUE THREAD

**Issue by sharwell** [dotnet/roslyn#34448]
**Version Used**: Visual Studio 2019 RC

**Steps to Reproduce**:

```csharp
object value = null;
foreach (var item in set) {
  if (item is null)
    continue;

  value ??= new object();
  Console.WriteLine(value.GetHashCode());
}
```

**Expected Behavior**:

IDE0059 is not reported for the declaration of `value`.

**Actual Behavior**:

IDE0059 is reported, and the code fix leaved the code not able to compile.

**Comment by mavasani** [comment]
I believe this is fixed.
